src/courses/views.py

```diff
@@ -214,15 +218,70 @@ class CourseStudentCreate(NonPublicOnlyViewMixin, SuccessMessageMixin, LoginRequ
     success_url = reverse_lazy('quests:quests')
     success_message = "You have been added to the %(course)s course"
 
+    def get(self, request, *args, **kwargs):
+        # when accessing this view, check if we need a form at all, or can just create the studentcourse object using all defaults
+        # if simplified_course_registration is enabled in siteconfig and all three form fields are hidden, object should be created automatically
+        simpleregistration = SiteConfig.get().simplified_course_registration
+        form = self.get_form()
+
+        form_auto_create_condition = (
+            simpleregistration and
+            form.fields['semester'].widget.input_type == 'hidden' and
+            form.fields['block'].widget.input_type == 'hidden' and
+            form.fields['course'].widget.input_type == 'hidden'
+        )
+
+        if form_auto_create_condition:
+            # form_auto_create_condition == True means only one active/current semester, block, course object
+            # initialize semester, block, course objects in variables to create new CourseStudent with
+            auto_create_semester = SiteConfig.get().active_semester
+            auto_create_block = Block.objects.filter(active=True).first()
+            auto_create_course = Course.objects.filter(active=True).first()
+
+            # before object is created, check for existing identical CourseStudent object to avoid duplicate error
+            duplicate_coursestudent = CourseStudent.objects.filter(
+                user=self.request.user,
+                semester=auto_create_semester,
+                block=auto_create_block,
+                course=auto_create_course
+            )
+
+            # if duplicate CourseStudent object exists, raise duplicate error
+            if duplicate_coursestudent.exists():
+                messages.error(self.request, "An identical course entry for this user already exists.")
```

Comment:
Since this will happen automatically, more feedback might be helpful in this message, like:
{{user}} is already registered in {{course}} course, in {{block}} group, for {{semester}} semester (as an example)

Reply:
pending the implementation of previously discussed changes involving the UserPassesTestMixin view and having illegal accesses of the student registration form return an Error 403, this specific logic change has not been implemented. If a work-around to still display the error message despite the niche method of raising it is desired, please let me know!

Reply:
Nah, thanks for checking. Can leave it as is.